src/mongoc/mongoc-client.c

```diff
@@ -1183,6 +1184,7 @@ _mongoc_client_command_simple_with_hint (mongoc_client_t           *client,
                                    command, NULL, read_prefs);
 
    cursor->hint = hint;
+   cursor->is_write_command = true;
```

Comment:
Yeah, this looked like a glaring bug. It might have snuck in when I was testing something manually and accidentally staged it. Sorry about that.